File: CS50/SET_1/Dinheiro.c
#include <math.h>
#include <stdio.h>

void cash();

int main(void)
{
    cash();
}

void cash()
{
    float debt = 0;
    do
    {
        printf("Troca devida: ");
        scanf("%f", &debt);
        getchar();
    }while (debt <= 0);

    int debt_cents = round(debt * 100);
    int qtd_coins = 0;

    do
    {
        if ((debt_cents - 25) >= 0)
        {
            qtd_coins++;
            debt_cents -= 25;
        }
        else if ((debt_cents - 10) >= 0)
        {
            qtd_coins++;
            debt_cents -= 10;
        }
        else if ((debt_cents - 5) >= 0)
        {
            qtd_coins++;
            debt_cents -= 5;
        }
        else
        {
            qtd_coins++;
            debt_cents -= 1;
        }
    }while (debt_cents > 0);

    printf("%i", qtd_coins);
}

File: CS50/SET_6/dinheiro.py
cash = float(input("Troca devida: "))

cash = round(cash * 100)

coins = 0

while cash > 0:
    match cash:
        case _ if cash >= 25:
            cash -= 25
            coins += 1
        case _ if cash >= 10:
            cash -= 10
            coins += 1
        case _ if cash >= 5:
            cash -= 5
            coins += 1
        case _ if cash >= 1:
            cash -= 1
            coins += 1

print(coins)